### ISSUE TITLE
DBeaver is freezing when executing a query with many parentheses in SELECT clause

### ISSUE THREAD

**Issue by vicreal** [dbeaver/dbeaver#13905]
**Windows 10 (64 bit) + DBeaver 21.2.0 + PostgreSQL 13** (tunnels or proxies not used).
Dbeaver hangs and becomes unresponsive when executing a query with many parentheses in the SELECT clause

**Example 1**:
`SELECT ((((((((((((((((5))))))))))))))))`

**Example 2**:
```
SELECT
	REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE
	(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(
	'abrakadabra', 
	CHR(1), '?'), CHR(2), '?'), CHR(3), '?'), CHR(4), '?'), CHR(5), '?'), CHR(6), '?'), CHR(7), '?'), CHR(8), '?'),
	CHR(11), '?'), CHR(12), '?'), CHR(14), '?'), CHR(15), '?'), CHR(16), '?'), CHR(17), '?'), CHR(18), '?'),
	CHR(19), '?'), CHR(20), '?'), CHR(21), '?'), CHR(22), '?'), CHR(23), '?'), CHR(24), '?'), CHR(25), '?'),
	CHR(26), '?'), CHR(27), '?'), CHR(28), '?'), CHR(29), '?'), CHR(30), '?'), CHR(31), '?') AS statement_body
```
**P.S**. The bug occurred [here](https://github.com/dbeaver/dbeaver/issues/10256) and [here](https://github.com/dbeaver/dbeaver/issues/9054), but it was [fixed](https://github.com/dbeaver/dbeaver/issues/9054#issuecomment-760913350) in version **7.3.4**. Apparently, it has now reappeared. So, I'm opening a new ticket...

**Comment by uslss** [comment]
thanks for the report

**Comment by LonwoLonwo** [comment]
Maybe release https://github.com/JSQLParser/JSqlParser/releases/tag/jsqlparser-4.2 will be helpful in this situation.

**Comment by jetzerb** [comment]
I ran across this issue today, but noticed an oddity.  If you include another statement _before_ the one with all the parens, there is no problem.  Some examples on `Version 21.3.2.202112271055`

```sql
select (((((((((((0)))))))))));
-- DBeaver hangs 7-10 seconds,
-- then pops up the "Executing Query" progress info and runs another 7-10 seconds
-- DBeaver reports "1 row(s) fetched" time does not include the "hang" time; reports 7-10 seconds

select (((((((((((0)))))))))));
select 1;
-- Same behavior as above

select 1;
select (((((((((((0)))))))))));
-- Instantaneous
-- DBeaver reports "1 row(s) fetched - 24ms"
```

**Comment by uslss** [comment]
fixed by #15155
